modules/nixvim/nixvim.nix

```
@@ -0,0 +1,31 @@
+{
+  config,
+  lib,
+  ...
+}: {
+  options.stylix.targets.nixvim.enable =
+    config.lib.stylix.mkEnableTarget "nixvim" true;
+
+  config = lib.mkIf config.stylix.targets.nixvim.enable {
```

What about the following assertion:

```suggestion
  config = lib.mkIf config.stylix.targets.nixvim.enable {
    assertions = [
      {
        assertion = options.programs?nixvim;

        message = ''
          'stylix.targets.nixvim.enable' is set to
          ${config.stylix.targets.nixvim.enable}, but the external
          'options.programs.nixvim' dependency is unavailable:
          https://github.com/nix-community/nixvim
        '';
      }
    ];
```

Note that I have not actually tested this.

---

With `mkIf`, evaluation still fails when the option doesn't exist, even if `config.stylix.targets.nixvim.enable` is set to `false`

---

The following should gracefully skip evaluation if `config.stylix.targets.nixvim` is not available:

```suggestion
  config = let
    targets = config.stylix.targets;
  in lib.mkIf (targets?nixvim.enable && targets.nixivm.enable) {
```

Note that I have not actually tested this.